### PR TITLE
Added arguments support for handlebars parsing

### PR DESCRIPTION
**Pull request by maxyushchenko**
* Fixed handlebars parser to pick up Randomizer along with datasets
* Added support for passing parameters to handlebars parser (including overloads)
* Added sample in documentation

### CODE REVIEW

**Comment by bchavez** [comment]
Hi Max,

Thank you for the pull request. I appreciate it.

Please give me some time to review the code changes and I'll get back to you.

Thanks,
Brian

:watch: :city_sunset: ***["I just can't wait... I just can't wait... for saturday night..."](https://www.youtube.com/watch?v=ISbXzQz__Rc)***

File: Source/Bogus/DataSet.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace Bogus
    /// <summary>
    /// Data set methods that access the BSON database of locales.
    /// </summary>
-   public class DataSet : ILocaleAware, IHasRandomizer
+   public class DataSet : ILocaleAware, IHasRandomizer, IDataset
```
Comment (bchavez) [inline Source/Bogus/DataSet.cs]: Cosmetic: `IDataset` should probably be named `IDataSet` to be consistent with the `DataSet` class.

**Comment by maxyushchenko** [comment]
Hi Brian,
Sure!
And thank you for a great lib :)

File: Source/Bogus/DataSet.cs (hunk 1 above)
Comment (maxyushchenko) [reply Source/Bogus/DataSet.cs]: Yep, missed it, thanks

**Comment by bchavez** [comment]
Hi Max,

Thanks again for the pull request. You did some amazing work here!

Your feature has been released and is now in [**Bogus `v22.1.1`**](https://www.nuget.org/packages/Bogus/22.1.1).

Enjoy! Have a great week!

Thanks,
Brian

:walking: :sunglasses: ***["So don't delay... act now supplies are running out..."](https://www.youtube.com/watch?v=LQj--Kjn0z8)***

**Comment by maxyushchenko** [comment]
Hi Brian,
Thank you for accepting this contribution and for the work you've done! :)

Kind regards,
Max

**Pull request merged**